ui-ngx/src/app/shared/components/entity/entity-subtype-select.component.ts

```diff
@@ -193,34 +194,39 @@ export class EntitySubTypeSelectComponent implements ControlValueAccessor, OnIni
     }
   }
 
-  displaySubTypeFn(subType?: EntitySubtype | string): string | undefined {
+  displaySubTypeFn(subType?: EntitySubtype | string | EntityInfoData): string | undefined {
     if (subType && typeof subType !== 'string') {
+      const typeName = this.isEntitySubType(subType) ? subType.type : subType.name;
       if (this.typeTranslatePrefix) {
-        return this.translate.instant(this.typeTranslatePrefix + '.' + subType.type);
+        return this.translate.instant(this.typeTranslatePrefix + '.' + typeName);
       } else {
-        return subType.type;
+        return typeName;
       }
     } else {
       return this.translate.instant('entity.all-subtypes');
     }
   }
 
-  findSubTypes(searchText?: string): Observable<Array<EntitySubtype | string>> {
+  findSubTypes(searchText: string): Observable<Array<EntitySubtype | string | EntityInfoData>> {
     return this.getSubTypes().pipe(
       map(subTypes => subTypes.filter( subType => {
-          return searchText ? (typeof subType === 'string' ? false : subType.type === searchText) : false;
+        if (typeof subType === 'string') {
+          return false;
+        } else {
+          return this.isEntitySubType(subType) ? subType.type : subType.name === searchText;
+        }
       }))
     );
   }
 
-  getSubTypes(): Observable<Array<EntitySubtype | string>> {
+  getSubTypes(): Observable<Array<EntitySubtype | string | EntityInfoData>> {
```

Comment:
Perhaps it is better to rewrite so the function returns only the string data type?